### PR TITLE
Implement better timetable layout

### PR DESCRIPTION
**Pull request by VojtaStanek**
The story is that few years ago I tried this tool (back when it was using mongoDB and tray wasn't there - which I apparently invented also independetly - just called it zásobník instead of odkladiště). I created my own tool, which is (at least in my eyes) better that this in some ways, but worse in others. The main difference being ease of use of the timetable. 

I have decided to retire my tool and contribute here. I did it as a opt-in option, since it's quite a big change. I have decided to use typescript, although it is not used in this project, because the source is written in TS and I saw at as the only option to manage to do it (since I am able to check the code for dumb mistakes easily). Sometimes the variables naming might be weird, it can be a leftover from my system (which used 3-levels of entities to represent a program). 

**Features of my implementation:** 
- Time guidelines (vertical lines).
- Zooming.
- Up to one-minute precision (depending on the zoom).
- Handles cases where we have to split a block across days or groups.

<details><summary>
<strong>Block splitting "bug"</strong>
</summary>
<p>

Following two images are showing the same program:

image

image

</p>
</details> 

<details><summary>
<strong>Video demo</strong>
</summary>
<p>

https://github.com/user-attachments/assets/b4c2b071-f5c6-4972-986c-fc906494a52f

</p>
</details> 

**Missing features:**
- Current time indicator 
- Filters on the top
- Printing 
- Program swap

**Commits:**
- Update README with local development setup instructions
- Add timetable layout version to settings
- Implement TimetableV2

ref #245, #77, #72

### CODE REVIEW

**Review by janzarsky — CHANGES_REQUESTED** [review]
Nice work! I have added some comments. I fixed some CI problems regarding formatting and linting - please rebase and fix the issues (or add exceptions).

My strategy for this new layout would be: 1) merge it soon, 2) keep it as separate option and work on it, 3) try to unify code between v1 and v2 where useful, 4) we will see how it goes.

File: src/components/Settings.jsx [hunk 1]
```diff
@@ -30,14 +30,27 @@ import { useGetProgramsQuery } from "../store/programsApi";
...
         {userLevel >= level.EDIT && timetableTitle && <TimetableTitle />}
-        {userLevel >= level.EDIT && <TimeStep />}
-        {userLevel >= level.EDIT && <Width />}
+        {timetableLayoutVersion === "v1" && (
+          <>
+            {userLevel >= level.EDIT && <TimeStep />}
+            {userLevel >= level.EDIT && <Width />}
+          </>
+        )}
+        {userLevel >= level.EDIT && <TimetableLayoutVersion />}
```
[the remark this answers is not in the document]
Comment (VojtaStanek) [reply src/components/Settings.jsx]: Done.

File: src/store/settingsApi.js [hunk 2]
```diff
@@ -4,11 +4,13 @@ import { streamingUpdatesEnabled } from "../helpers/StreamingUpdates";
 
 export const DEFAULT_TIME_STEP = 15 * 60 * 1000;
 export const DEFAULT_WIDTH = 100;
+export const DEFAULT_TIMETABLE_LAYOUT_VERSION = "v1";
 
 function addDefaults(data) {
   return {
     timeStep: DEFAULT_TIME_STEP,
     width: DEFAULT_WIDTH,
+    timetableLayoutVersion: DEFAULT_TIMETABLE_LAYOUT_VERSION,
```
[the remark this answers is not in the document]
Comment (VojtaStanek) [reply src/store/settingsApi.js]: Done.

File: src/components/App.jsx [hunk 3]
```diff
@@ -256,20 +257,28 @@ export function TimetableWrapper({
   permissionsLoaded,
   printView = false,
 }) {
+  const { table } = useSelector((state) => state.auth);
+  const { data: settings, isSuccess: settingsLoaded } = useGetSettingsQuery(table);
```
[the remark this answers is not in the document]
Comment (VojtaStanek) [reply src/components/App.jsx]: Done.

File: src/components/Settings.jsx [hunk 4]
```diff
@@ -25,17 +25,40 @@ import {
...
-        {userLevel >= level.EDIT && <TimeStep />}
-        {userLevel >= level.EDIT && <Width />}
+        {layoutVersion === "v1" && (
+          <>
+            {userLevel >= level.EDIT && <TimeStep />}
+            {userLevel >= level.EDIT && <Width />}
+          </>
+        )}
+        {userLevel >= level.EDIT &&
+          config.timetableLayoutVersionSwitchingEnabled && (
```
Comment (janzarsky) [inline src/components/Settings.jsx:59]: why not.. but I have two problems with this approach:
1) this should go ideally into separate PR
2) I need to be able to toggle this in tests, this is the whole motivation behind config slice and override config reducer. So components should not be importing config directly but rather use config slice. Or some other method - I don't care. But I need to be able to write tests for both flag on/off, see here for example: https://github.com/janzarsky/scout-planner/blob/c0a81e029bb4f7c1cc7f09e83ffa09bd0a4fdc5c/cypress/component/EditProgramModal.cy.jsx#L80

so.. maybe leave this for a separate PR and just use the old method for now?

**Review by janzarsky — CHANGES_REQUESTED** [review]

**Comment by janzarsky** [comment]
and one more thing: please remove fix issue references from the PR description, maybe just mention those, since it will not be truly fixed until the new layout will be by default

**Comment by VojtaStanek** [comment]
> and one more thing: please remove fix issue references from the PR description, maybe just mention those, since it will not be truly fixed until the new layout will be by default

makes sense, done.

File: src/components/Settings.jsx (hunk 4 above)
Comment (VojtaStanek) [reply src/components/Settings.jsx]: I prepared two separate PRs: #730 and #731

File: src/components/TimetableV2.tsx [hunk 5]
```diff
@@ -0,0 +1,1176 @@
...
+  const [ownerFilter, setOwnerFilter] = useState<string | null>(null);
+  const people = usePeople();
+  const availableOwners = useMemo(() => {
+    const result: { id: string; name: string; count: number }[] = [];
+    for (const program of programs) {
+      for (const { person: ownerId } of program.people) {
+        let ownerRecord = result.find((it) => it.id === ownerId);
+        if (!ownerRecord) {
+          const person = people.find((it) => it._id === ownerId);
+          ownerRecord = { id: ownerId, name: person!.name, count: 0 };
```
Comment (janzarsky) [inline src/components/TimetableV2.tsx:572]: Looks like this is failing due to data inconsistency. There is no guarantee that `people.find(...)` will succeed, so there needs to be some fallback in `person!.name`.

Comment (janzarsky) [inline src/components/TimetableV2.tsx:572]: Looks like this is failing due to data inconsistency. There is no guarantee that `people.find(...)` will succeed, so there needs to be some fallback in `person!.name`.

Comment (janzarsky) [inline src/components/TimetableV2.tsx:572]: Looks like this is failing due to data inconsistency. There is no guarantee that `people.find(...)` will succeed, so there needs to be some fallback in `person!.name`.

**Review by janzarsky — APPROVED** [review]
Nice! lets merge this. Found one small issue that can be fixed later.

**Pull request merged**